aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -100,6 +115,67 @@ public void deleteFile(String path) {
     client().deleteObject(deleteRequest);
   }
 
+  /**
+   * Deletes the given paths in a batched manner.
+   * <p>
+   * The paths are grouped by bucket, and deletion is triggered when we either reach the configured batch size
+   * or have a final remainder batch for each bucket.
+   *
+   * @param paths paths to delete
+   */
+  @Override
+  public void deleteFiles(Iterable<String> paths) {
```

Comment:
If the batch size is 1, I think this should fall back to the normal delete implementation.

Reply:
We can. The current logic is generic but for the case where batch size is 1 there's more unnecessary state management, so I'm in favor of falling back to just looping and calling the existing delete.

Reply:
Actually, one issue with falling back to the normal delete implementation in case the batch size is 1 is we currently don't have any expectation on the exception being surfaced by FileIo#deleteFile in case of a failure during deletion.
 
With deleteFiles we are keeping track of failures and then surfacing them at the end, so in the case the batch size is 1 we would have to catch generic Exceptions and keep track of failures that way which seems more messy. So now I'm thinking it makes more sense to keep the generic batch removal logic (the multi-map state management is light-weight). Let me know your thoughts @rdblue